### ISSUE TITLE
Please ship the envoy "tools"

### ISSUE THREAD

**Issue by antifuchs** [tetratelabs/getenvoy#85]
**Is your feature request related to a problem? Please describe.**

I'm trying to upgrade a configuration file from an envoy 1.5.0 install to be compatible with version 1.13.1; of course, valid keys got renamed/deprecated/removed. My iteration time would be far faster (and my deployments far safer!) if I could use the [config load check tool](https://www.envoyproxy.io/docs/envoy/latest/install/tools/config_load_check_tool) that envoy has in its source tree - but none of the binary distributions of envoy ship it; I have to build it myself, in a process that's kinda slow.

**Describe the solution you'd like**

Please build and ship the tools, especially `config_load_check_tool` as an additional installation target, e.g. `config_load_check_tool:1.13.1/linux-glibc` similar to `standard:1.13.1/linux-glibc`.

**Describe alternatives you've considered**

Another approach would be to behave like the `rustup` program, in which users can choose "toolchains" that automatically get installed alongside each version; those are usually supporting programs or data files. I don't think the set of additional tools warrants having this complexity, especially as there is already a separate namespace that you could use for this.

**Additional context**

The way I imagine I'd use this is in a staged docker build toolchain: 
1. In the "build" container, download `getenvoy`, install the standard binary and config load check tool.
2. Copy the configuration into the build container and validate it with the tool.
3. Copy the envoy binary and config into the "final" container and deploy that.

**Comment by antifuchs** [comment]
.......and today is the day that I learn `envoy --mode validate` exists and does what I need.

I do think there is a use case for the *other* tool (validating the message schema and checking routes), so I'll leave this issue open.

**Comment by codefromthecrypt** [comment]
can you identify other tools? otherwise we should close this issue because open-ended issues without specific outcomes usually never complete

**Comment by antifuchs** [comment]
They're listed in the documentation section ["Tools > Tools"](https://www.envoyproxy.io/docs/envoy/latest/operations/tools/tools). Currently, that list is:

* [`config_load_check_tool`](https://www.envoyproxy.io/docs/envoy/latest/operations/tools/config_load_check_tool)
* [`router_check_tool `](https://www.envoyproxy.io/docs/envoy/latest/operations/tools/route_table_check_tool)
* [`schema_validator_tool`](https://www.envoyproxy.io/docs/envoy/latest/operations/tools/schema_validator_check_tool)

And, optionally, the generated configuration from the [config_generator](https://www.envoyproxy.io/docs/envoy/latest/operations/tools/config_generator) section.

In short, I'm of the opinion that if you document the existence of a thing that allows debugging a deployment, you should include it in the binary distribution used to create the deployment.

**Comment by codefromthecrypt** [comment]
ok so basically IIRC you would like the packaging to include more tools, as it currently only includes the binary. This is less about get envoy the CLI and more about the assets downloaded for the version requested, correct?

**Comment by antifuchs** [comment]
If you read the docs you'll find that these are not assets: They're build products, same as the `envoy` binary that's shipped in the various downloadable distributions. In order to get, for example, the `router_check_tool`, you have to download the source code for envoy that you're deploying, get a working build system for the environment you're deploying in, then build that tool and then ship it to the place you're deploying, and only then can you run it there.

If you've no interest in improving this situation for your users, feel free to close - I've no skin in the game here anymore, having moved away from Envoy 1.5 years ago, and now much happier for it.

**Comment by codefromthecrypt** [comment]
@antifuchs sorry please don't mistake a poor choice of vocab for anything. By assets, I just meant files hosted (which would include envoy binary). I've been trying to cleanup issues and understand things, and no one answered you so figured worth talking to you about it, and understand. Sorry I am new to Envoy so I don't know the proper vocab for the files, I wasn't trying to dismiss, rather understand.

For context, there's an effort to migrate the "builds" and so I was hoping to identify which files (products assets etc) should be migrated upstream, to ensure they are built by upstream also. This repo doesn't do builds at all, so I needed to understand to route the issue to the right repo.

As of your last comment, I fully understand what you mean and sorry I didn't understand quicker, but hey you saved the next person 😎 so thanks!
